### PR TITLE
Backport PR #12258 on branch 3.4.x (Default is no virtual rendering + Relax virtual notebook rendering on idle)

### PR DESCRIPTION
**Pull request by echarles**
Backport PR https://github.com/jupyterlab/jupyterlab/pull/12258: Default is no virtual rendering + Relax virtual notebook rendering on idle

Fixes https://github.com/jupyterlab/jupyterlab/pull/12166

### CODE REVIEW

**Comment by echarles** [comment]
@fcollonval The sensible CI flows (ui regression, js-notebook) are OK. The other failing flows are not related to these changes.

This is ready for review/merge.

File: packages/notebook/src/widget.ts [hunk 1]
```diff
@@ -190,6 +191,7 @@ export class StaticNotebook extends Widget {
     this.node.dataset[UNDOER] = 'true';
     this.node.dataset[CODE_RUNNER] = 'true';
     this.rendermime = options.rendermime;
+    this.translator = options.translator || nullTranslator;
```
Comment (fcollonval) [inline packages/notebook/src/widget.ts:194]: You should pass the translator when creating the widget in `packages/notebook/src/widgetfactory.ts#L98` - you can get it within the factory as `this.translator`.

You should prefer the nullish operator to the or operator:

```suggestion
    this.translator = options.translator ?? nullTranslator;
```

**Comment by fcollonval** [comment]
I would like to raise the target for this at tomorrow's meeting. As this may change the loading time for the user, I would like others opinions.

**Comment by echarles** [comment]
@fcollonval Can we merge this in 3.3.x or should we target 3.4.x?

**Comment by fcollonval** [comment]
We will target 3.4.0 - I'm doing a 3.3.x release then I'm taking care of the 3.4.x boot up

**Comment by echarles** [comment]
Thx @fcollonval  for targeting 3.4.x. CI remains ok.

**Comment by echarles** [comment]
👍 @fcollonval

**Pull request merged**